### ISSUE TITLE
False positives with `no-array-prototype-extensions` rule

### ISSUE THREAD

**Issue by gilest** [ember-cli/eslint-plugin-ember#1537]
Further to #1534 there are additional false positives when this rule is run against [ember-file-upload](https://github.com/adopted-ember-addons/ember-file-upload/runs/7440106845?check_suite_focus=true).

- `.clear()` on a `Set` (probably on a Map, too)
- `RSVP.reject()`

If I have time tomorrow I'll open a PR to address.

**Comment by bmish** [comment]
Thanks a lot for this report! Ignoring `RSVP` should be easy using the same technique as the last fix:

https://github.com/ember-cli/eslint-plugin-ember/blob/42a3e857fbca6168ecb8c5e4378f760cb8307079/lib/rules/no-array-prototype-extensions.js#L60

For `Map` / `Set` / `WeakMap` / `WeakSet` / or the [tracked-built-ins](https://github.com/tracked-tools/tracked-built-ins) versions of those, we'll need to come up with some heuristics to detect when the variables are those data types. Some of my  ideas:
* Ignore variables with "map" or "set" in the name
* Ignore variables where we can statically determine their value or that at least one of their initializations was to a map/set

**Comment by bmish** [comment]
I began a fix with one of the heuristics in #1538.

**Comment by gilest** [comment]
Attempted to cover the remaining case in #1539

Would appreciate your review @bmish

**Comment by bmish** [comment]
I'm reopening this because there are still some improvements needed to cover the [original scenario](https://github.com/adopted-ember-addons/ember-file-upload/blob/afc294a25f2ca82dafb0da181debd1f34584c6a9/ember-file-upload/src/queue.ts#L180):

1. Class property definitions

```js
class MyClass {
  foo = new Set();
  myFunc() { foo.clear() }
}
```

2. TypeScript with private identifiers

```ts
class MyClass {
  #foo: Set<UploadFile> = new Set();
  myFunc() {
    #foo.clear();
  }
}
```

I'll try to look into why my solution in #1538 doesn't handle those in the coming day or so.

**Comment by bmish** [comment]
@gilest all the necessary fixes should now be available in [v11.0.3](https://github.com/ember-cli/eslint-plugin-ember/releases/tag/v11.0.3). Let me know if you have any other issues. Thanks for testing and reporting!

**Comment by gilest** [comment]
Awesome work @bmish 🙌🏻
